external/external.go

```diff
@@ -0,0 +1,242 @@
+		"returnValue":         json.Number(strconv.Itoa(e.ReturnValue)),
+		"args":                argsRef,
+		"stackAddresses":      stackAddressesRef,
+	}, nil
+}
+
+func jsonRoundTripArgumentValue(v interface{}) (interface{}, error) {
+	m, err := json.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+	buf := bytes.NewBuffer(m)
+
+	var u interface{}
+	d := json.NewDecoder(buf)
+	d.UseNumber()
+	err = d.Decode(&u)
+	if err != nil {
+		return nil, err
+	}
+	return u, nil
+}
+
+// Argument holds the information for one argument
+type Argument struct {
+	ArgMeta
+	Value interface{} `json:"value"`
+}
+
+// ArgMeta describes an argument
+type ArgMeta struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (arg *Argument) UnmarshalJSON(b []byte) error {
+	type argument Argument //alias Argument so we can unmarshal it within the unmarshaler implementation
+	d := json.NewDecoder(bytes.NewReader(b))
+	d.UseNumber()
+	if err := d.Decode((*argument)(arg)); err != nil {
+		return err
+	}
+	if arg.Value == nil {
+		return nil
+	}
+	if num, isNum := arg.Value.(json.Number); isNum {
+		switch arg.Type {
+		case "int", "pid_t", "uid_t", "gid_t", "mqd_t", "clockid_t", "const clockid_t", "key_t", "key_serial_t", "timer_t", "landlock_rule_type":
+			tmp, err := strconv.ParseInt(num.String(), 10, 32)
+			if err != nil {
+				return err
+			}
+			arg.Value = int32(tmp)
+		case "long":
+			tmp, err := num.Int64()
+			if err != nil {
+				return err
+			}
+			arg.Value = tmp
+		case "unsigned int", "u32", "mode_t", "dev_t":
+			tmp, err := strconv.ParseUint(num.String(), 10, 32)
+			if err != nil {
+				return err
+			}
+			arg.Value = uint32(tmp)
+		case "unsigned long", "u64", "off_t", "size_t", "void*", "const void*":
+			tmp, err := strconv.ParseUint(num.String(), 10, 64)
+			if err != nil {
+				return err
+			}
+			arg.Value = uint64(tmp)
+		case "float":
+			tmp, err := strconv.ParseFloat(num.String(), 32)
+			if err != nil {
+				return err
+			}
+			arg.Value = float32(tmp)
+		case "float64":
+			tmp, err := num.Float64()
+			if err != nil {
+				return err
+			}
+			arg.Value = tmp
+		case "unsigned short", "old_uid_t", "old_gid_t":
+			tmp, err := strconv.ParseUint(num.String(), 10, 16)
+			if err != nil {
+				return err
+			}
+			arg.Value = uint16(tmp)
+		default:
+			return fmt.Errorf("unrecognized argument type")
+		}
+	}
+	if arg.Type == "const char*const*" || arg.Type == "const char**" {
+		argValue := arg.Value.([]interface{})
+		tmp := make([]string, len(argValue))
+		for i, v := range argValue {
+			tmp[i] = fmt.Sprint(v)
+		}
+		arg.Value = tmp
+	}
+	return nil
+}
+
+// SlimCred struct is a slim version of the kernel's cred struct
+// it is used to unmarshal binary data and therefore should match (bit by bit) to the `slim_cred_t` struct in the ebpf code.
+// ANY CHANGE TO THIS STRUCT WILL BE REQUIRED ALSO TO bufferdecoder.SlimCred
```

Comment:
please update here https://github.com/aquasecurity/tracee/blob/main/tracee-ebpf/tracee/internal/bufferdecoder/protocol.go#L91 that changes must be done to both `SlimCred` structs: both in `external/external.go` and in `pkg/external/external.go`. 
Are we planning eventually to remove `pkg/external` with this https://github.com/aquasecurity/tracee/pull/1455?

Reply:
Yes we will merge those two one after the other.
I removed the old external in #1455 thanks for noticing.